testing/mock.ts

```diff
@@ -341,19 +358,22 @@ export interface Stub<
 /** Replaces an instance method with a Stub. */
 export function stub<
   Self,
-  // deno-lint-ignore no-explicit-any
-  Args extends unknown[] = any[],
-  Return = undefined,
->(self: Self, property: keyof Self): Stub<Self, Args, Return>;
+  Prop extends keyof Self,
+>(
+  self: Self,
+  property: Prop,
+): Stub<Self, GetParametersFromProp<Self, Prop>, GetReturnFromProp<Self, Prop>>;
 export function stub<
   Self,
-  Args extends unknown[],
-  Return,
+  Prop extends keyof Self,
 >(
   self: Self,
-  property: keyof Self,
-  func: (this: Self, ...args: Args) => Return,
-): Stub<Self, Args, Return>;
+  property: Prop,
+  func: (
+    this: Self,
+    ...args: Partial<GetParametersFromProp<Self, Prop>>
```

Comment:
I think this should just be unknown[] since partial won't work. That would allow it to be replaced with a function that doesn't need any specific arguments.

Reply:
That sounds fine by me, as I don't have to fix the type error then xD
It seems like it should be `any[]` even, otherwise the `stub()` call in the "assertSpyCallAsync method" test gives a type error saying that the `x?: number` and `y?: number` parameters are not assignable to `unknown`.

The downside of this is that like that there is no type checking performed at all when using the `stub` function.

Reply:
The returned stub has its Args derived from the prop on the instance. So calling the stub function would still have type checking and so would the assertions on the stub calls.

The only thing that wouldn't have type checking is the func arg, which didn't have type checking before either. Before your proposed change, the stub could only be called directly with the same args as the func had.

Reply:
Ah yeah good point. Fwiw, type checking did seem to work with the Partial type in place. But that obviously meant no other function signatures were allowed. So `any[]` seems like the better option here.